functions/src/index.ts

```diff
@@ -0,0 +1,91 @@
+import * as functions from "firebase-functions";
+import * as admin from "firebase-admin";
+import { Client as NotionClient } from "@notionhq/client";
+
+admin.initializeApp();
+const db = admin.firestore();
+
+const notionSecret = process.env.NOTION_SECRET
+const databaseId = process.env.NOTION_DATABASE_ID
+
+
+if (!notionSecret || !databaseId) {
+  throw new Error("Environment variables for Notion are not set.");
+}
+
+const notionClient = new NotionClient({
+  auth: notionSecret,
+});
+
+
+//export const syncNotionEventsToFirestore = functions.https.onRequest(async (_, response) => {
+export const scheduledSyncNotionEventsToFirestore = functions.pubsub.schedule('0 * * * *').onRun(async (context) => {
+	try {
+		if (!databaseId) {
+			throw new Error("Environment variable NOTION_DATABASE_ID is not set.");
+		}
+
+		const notionResponse = await notionClient.databases.query({ database_id: databaseId });
+
+		notionResponse.results.forEach(async (page: any) => {
+			const blockResponse = await notionClient.blocks.children.list({
+				block_id: page.id,
+			});
+
+			const date = page.properties["Date"]?.date?.start ?? "";
+			const eventDate = date ? new Date(date) : null; 
+
+			if (!eventDate || eventDate < new Date()) {
+				return;
+			}
+
+			const isInPersonOrOnline = page.properties["In Person/Online"]?.select?.name ?? null;
+			const hasConfirmedRoom = page.properties["Confirmed Room"]?.multi_select.length > 0;
+			//const hasTypeformSignIn = page.properties["Typeform Sign In"]?.url ?? null;
+			//const hasTypeformFeedback = page.properties["Typeform Feedback"]?.url ?? null;
+
+			if (!isInPersonOrOnline || !hasConfirmedRoom) {
+				return;
+			}
+
+			let description = "";
+
+			for (const block of blockResponse.results) {
+
+				if ('type' in block && block.type === "paragraph" && block.paragraph.rich_text.length > 0) {
+
+					const firstNonEmptyParagraph = block.paragraph.rich_text.find(rt => rt.plain_text.trim() !== "");
+					if (firstNonEmptyParagraph) {
+						description = firstNonEmptyParagraph.plain_text;
+						break; 
+					}
+				}
+			}
+
+			const title = page.properties["Name"]?.title[0]?.plain_text ?? "Untitled Event";
```

Comment:
Now that I go through this again. I wouldn't put the event in Firebase if it doesn't have a title. It means that it is not ready yet.

Reply:
If you use `?? "Untitled Event""` then title will never be undefined, would it? If LHS gets a falsy value at any point then title gets assigned RHS value "Untitled Event"